pints/tests/test_mcmc_controller.py

```diff
@@ -1838,6 +1838,121 @@ def test_multi_chain_with_nones(self):
+        xs = pints.sample_initial_points(self.log_posterior, nchains)
+        self.assertEqual(len(xs), nchains)
+        [self.assertTrue(np.isfinite(self.log_posterior(x))) for x in xs]
+
+        # check parallel initialisation works
+        xs = pints.sample_initial_points(self.log_posterior, nchains,
+                                         parallel=True)
+        self.assertEqual(len(xs), nchains)
+        xs = pints.sample_initial_points(self.log_posterior, nchains,
+                                         parallel=True, n_workers=2)
+        self.assertEqual(len(xs), nchains)
+
+    def test_errors(self):
+        # tests errors when calling method with wrong inputs
+
+        # pass a non-callable object as random_sampler
+        nchains = 4
+        self.assertRaises(ValueError, pints.sample_initial_points,
+                          self.log_posterior, nchains,
+                          [0.015, 500, 10] * nchains)
+
+        # try non log-posterior without passing random_sampler
+        log_pdf = pints.toy.GaussianLogPDF()
+        self.assertRaises(ValueError, pints.sample_initial_points,
+                          log_pdf, nchains)
+
+        # n_chains < 1?
+        self.assertRaises(ValueError, pints.sample_initial_points,
+                          self.log_posterior, 0.5)
+
+    def test_bespoke_initialisation(self):
+        # test using user-specified initialisation function
+
+        # test that different initialisation produces different starting dist
+        nchains = 4
+        noise = 10
+        xs = pints.sample_initial_points(self.log_posterior, nchains)
+        log_prior1 = pints.UniformLogPrior(
+            [0.0199, 599.99, noise * 99.99],
+            [0.02, 600, noise * 100]
+        )
+        xs1 = pints.sample_initial_points(self.log_posterior, nchains,
+                                          log_prior1.sample)
+        self.assertTrue(sum(np.vstack(xs).mean(axis=0) <=
+                            np.vstack(xs1).mean(axis=0)) == 3)
+        [self.assertTrue(np.isfinite(self.log_posterior(x))) for x in xs]
+
+        # test initialisation for log_pdf (non-log-posterior)
+        log_pdf = pints.toy.GaussianLogPDF()
+        log_pdf1 = pints.toy.GaussianLogPDF(mean=[1, 1], sigma=[2, 2])
+        init_sampler = log_pdf1.sample
+        xs = pints.sample_initial_points(log_pdf, nchains, init_sampler)
+        [self.assertTrue(np.isfinite(log_pdf(x))) for x in xs]
+
+    def test_initialisation_fails(self):
+        # tests that initialisation can fail in specified number of tries
+
+        from scipy.stats import multivariate_normal
+        noise = 10
+        nchains = 4
+
+        def init_sampler(n_chains):
+            return multivariate_normal.rvs(mean=[0.015, 500, noise],
+                                           cov=np.diag([10, 10000, noise]),
+                                           size=nchains)
+
+        self.assertRaises(RuntimeError, pints.sample_initial_points,
```

Comment:
Maybe use an `assertRaisesRegex` here to check it's the correct RuntimeError?

Reply:
Done